src/games/strategy/engine/framework/GameRunner2.java

```diff
@@ -151,11 +158,18 @@ private static void usage() {
         + "   to start a server, you can optionally password protect the game using triplea.server.password=foo");
   }
 
-  public static void main(final String[] args) {
+  public static void main(String[] args) {
+    if( args == null ) {
+      args = new String[0];
+    }
+    if( exceptionHandler == null ) {
+      exceptionHandler = new ErrorHandler();
+    }
+    Thread.setDefaultUncaughtExceptionHandler( exceptionHandler);
+
     setupLogging();
     Console.getConsole().displayStandardError();
     Console.getConsole().displayStandardOutput();
-    System.setProperty("sun.awt.exception.handler", ErrorHandler.class.getName());
```

Comment:
System.setProperty("sun..") simply does not work. It does nothing as far as I can tell. I threw exceptions explicitly from various parts of the code and the error handler was never invoked.

So the System.setProperty did not apply to all threads, it applied to nothing.

Thread.setDefault( .. ) is a static method so it does not necessarily have a concept of the current thread other than what the internal implementation does.

In practice I've seen Thread.setDefault(...)  catch swing exceptions. There are some threads in unit tests that we spawn, generic Runnables, that look like they still might not get caught properly. Either way, if it catches everything or not, since it does catch more than we do now, we get benefit.